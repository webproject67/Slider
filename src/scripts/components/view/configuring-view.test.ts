import ConfiguringView from './configuring-view';
import Model from '../model/model';

const main = document.createElement('div');
main.id = 'banana';

const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  progress: true,
  range: 'one',
  draft: 0,
  start: 1,
  configuring: 0,
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'horizontal',
};

const model = new Model(main, state);
const configuringView = new ConfiguringView();

describe('snapshot', () => {
  test('renders correctly element1', () => {
    expect(new ConfiguringView().getElement(model)).toMatchSnapshot();
  });

  test('renders correctly element2', () => {
    model.setValue(['range'], ['range']);
    expect(new ConfiguringView().getElement(model)).toMatchSnapshot();
  });

  test('renders correctly element3', () => {
    model.setValue(
      ['from', 'to', 'view', 'flag', 'scale', 'progress'],
      [10, 90, 'vertical', false, false, false],
    );
    expect(new ConfiguringView().getElement(model)).toMatchSnapshot();
  });
});

describe('handleInputChange', () => {
  test('spyOn input change', () => {
    const somethingSpy = jest.spyOn(configuringView, 'handleInputChange');
    let evt: any;
    configuringView.handleInputChange(model, evt);
    expect(somethingSpy).toHaveBeenCalledTimes(1);
  });
});
