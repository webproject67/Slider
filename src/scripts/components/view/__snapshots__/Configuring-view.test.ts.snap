// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly element1 1`] = `
<div
  class="slider__labels"
>
  
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Минимальное значение
        
    <input
      class="slider__min"
      data-name="min"
      type="number"
      value="0"
    />
    
      
  </label>
  
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Максимальное значение
        
    <input
      class="slider__max"
      data-name="max"
      type="number"
      value="100"
    />
    
      
  </label>
  
      
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Текущее значение
        
    <input
      class="slider__to"
      data-name="to"
      readonly=""
      type="number"
      value="100"
    />
    
      
  </label>
  
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Шаг
        
    <input
      class="slider__step"
      data-name="step"
      type="number"
      value="1"
    />
    
      
  </label>
  
      
  <div
    class="slider__radio"
  >
    
        
    <label
      class="slider__label"
    >
      Горизонтальный
          
      <input
        checked=""
        class="slider__view"
        data-name="view"
        name="view7441"
        type="radio"
        value="horizontal"
      />
      
        
    </label>
    
        
    <label
      class="slider__label"
    >
      Вертикальный
          
      <input
        class="slider__view"
        data-name="view"
        name="view7441"
        type="radio"
        value="vertical"
      />
      
        
    </label>
    
      
  </div>
  
      
  <div
    class="slider__radio"
  >
    
        
    <label
      class="slider__label"
    >
      Одиночное значение
          
      <input
        checked=""
        class="slider__range"
        data-name="range"
        name="range1190"
        type="radio"
        value="one"
      />
      
        
    </label>
    
        
    <label
      class="slider__label"
    >
      Интервал
          
      <input
        class="slider__range"
        data-name="range"
        name="range1190"
        type="radio"
        value="range"
      />
      
        
    </label>
    
      
  </div>
  
      
  <label
    class="slider__label"
  >
    Значение
        
    <input
      checked=""
      class="slider__flag-checkbox"
      data-name="flag"
      type="checkbox"
    />
    
      
  </label>
  
      
  <label
    class="slider__label"
  >
    Шкала
        
    <input
      checked=""
      class="slider__scale-checkbox"
      data-name="scale"
      type="checkbox"
    />
    
      
  </label>
  
      
  <label
    class="slider__label"
  >
    Прогресс
        
    <input
      checked=""
      class="slider__progress"
      data-name="progress"
      type="checkbox"
    />
    
      
  </label>
  
    
</div>
`;

exports[`snapshot renders correctly element2 1`] = `
<div
  class="slider__labels"
>
  
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Минимальное значение
        
    <input
      class="slider__min"
      data-name="min"
      type="number"
      value="0"
    />
    
      
  </label>
  
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Максимальное значение
        
    <input
      class="slider__max"
      data-name="max"
      type="number"
      value="100"
    />
    
      
  </label>
  
      
        
  <label
    class="slider__label slider__label_state_displayed"
  >
    От
          
    <input
      class="slider__from"
      data-name="from"
      readonly=""
      type="number"
      value="0"
    />
    
        
  </label>
  
      
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    До
        
    <input
      class="slider__to"
      data-name="to"
      readonly=""
      type="number"
      value="100"
    />
    
      
  </label>
  
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Шаг
        
    <input
      class="slider__step"
      data-name="step"
      type="number"
      value="1"
    />
    
      
  </label>
  
      
  <div
    class="slider__radio"
  >
    
        
    <label
      class="slider__label"
    >
      Горизонтальный
          
      <input
        checked=""
        class="slider__view"
        data-name="view"
        name="view9961"
        type="radio"
        value="horizontal"
      />
      
        
    </label>
    
        
    <label
      class="slider__label"
    >
      Вертикальный
          
      <input
        class="slider__view"
        data-name="view"
        name="view9961"
        type="radio"
        value="vertical"
      />
      
        
    </label>
    
      
  </div>
  
      
  <div
    class="slider__radio"
  >
    
        
    <label
      class="slider__label"
    >
      Одиночное значение
          
      <input
        class="slider__range"
        data-name="range"
        name="range4666"
        type="radio"
        value="one"
      />
      
        
    </label>
    
        
    <label
      class="slider__label"
    >
      Интервал
          
      <input
        checked=""
        class="slider__range"
        data-name="range"
        name="range4666"
        type="radio"
        value="range"
      />
      
        
    </label>
    
      
  </div>
  
      
  <label
    class="slider__label"
  >
    Значение
        
    <input
      checked=""
      class="slider__flag-checkbox"
      data-name="flag"
      type="checkbox"
    />
    
      
  </label>
  
      
  <label
    class="slider__label"
  >
    Шкала
        
    <input
      checked=""
      class="slider__scale-checkbox"
      data-name="scale"
      type="checkbox"
    />
    
      
  </label>
  
      
  <label
    class="slider__label"
  >
    Прогресс
        
    <input
      checked=""
      class="slider__progress"
      data-name="progress"
      type="checkbox"
    />
    
      
  </label>
  
    
</div>
`;

exports[`snapshot renders correctly element3 1`] = `
<div
  class="slider__labels"
>
  
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Минимальное значение
        
    <input
      class="slider__min"
      data-name="min"
      type="number"
      value="0"
    />
    
      
  </label>
  
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Максимальное значение
        
    <input
      class="slider__max"
      data-name="max"
      type="number"
      value="100"
    />
    
      
  </label>
  
      
        
  <label
    class="slider__label slider__label_state_displayed"
  >
    От
          
    <input
      class="slider__from"
      data-name="from"
      readonly=""
      type="number"
      value="10"
    />
    
        
  </label>
  
      
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    До
        
    <input
      class="slider__to"
      data-name="to"
      readonly=""
      type="number"
      value="90"
    />
    
      
  </label>
  
      
  <label
    class="slider__label slider__label_state_displayed"
  >
    Шаг
        
    <input
      class="slider__step"
      data-name="step"
      type="number"
      value="1"
    />
    
      
  </label>
  
      
  <div
    class="slider__radio"
  >
    
        
    <label
      class="slider__label"
    >
      Горизонтальный
          
      <input
        class="slider__view"
        data-name="view"
        name="view5597"
        type="radio"
        value="horizontal"
      />
      
        
    </label>
    
        
    <label
      class="slider__label"
    >
      Вертикальный
          
      <input
        checked=""
        class="slider__view"
        data-name="view"
        name="view5597"
        type="radio"
        value="vertical"
      />
      
        
    </label>
    
      
  </div>
  
      
  <div
    class="slider__radio"
  >
    
        
    <label
      class="slider__label"
    >
      Одиночное значение
          
      <input
        class="slider__range"
        data-name="range"
        name="range6980"
        type="radio"
        value="one"
      />
      
        
    </label>
    
        
    <label
      class="slider__label"
    >
      Интервал
          
      <input
        checked=""
        class="slider__range"
        data-name="range"
        name="range6980"
        type="radio"
        value="range"
      />
      
        
    </label>
    
      
  </div>
  
      
  <label
    class="slider__label"
  >
    Значение
        
    <input
      class="slider__flag-checkbox"
      data-name="flag"
      type="checkbox"
    />
    
      
  </label>
  
      
  <label
    class="slider__label"
  >
    Шкала
        
    <input
      class="slider__scale-checkbox"
      data-name="scale"
      type="checkbox"
    />
    
      
  </label>
  
      
  <label
    class="slider__label"
  >
    Прогресс
        
    <input
      class="slider__progress"
      data-name="progress"
      type="checkbox"
    />
    
      
  </label>
  
    
</div>
`;
